torch/utils/data/_utils/collate.py

```diff
@@ -30,7 +30,11 @@ def default_convert(data):
     elif isinstance(data, tuple) and hasattr(data, '_fields'):  # namedtuple
         return elem_type(*(default_convert(d) for d in data))
     elif isinstance(data, collections.abc.Sequence) and not isinstance(data, string_classes):
-        return [default_convert(d) for d in data]
+        try:
+            return elem_type(default_convert(d) for d in data)
+        except BaseException:
```

Comment:
I think `BaseException` is too general. Let's use `TypeError` here.

Reply:
But what if the exception is different? It depends on the actual implementation, what it tries to do with the argument.

Reply:
I think `TypeError` is appropriate based on https://docs.python.org/3/library/exceptions.html#TypeError.

We can't guarantee how users implement each class, but we should advocate the right coding convention. And, if we set Error too loose in this case, users may get unexpected result. For example, if users' class does accept generator as input, it would also raise ValueError based on implementation. In this case, we suppress the Error and fall back to the list. But, users' expectation would be getting a subclass rather than a list.